src/meta/src/model/stream.rs

```diff
@@ -115,6 +115,18 @@ pub struct TableFragments {
 
     /// The parallelism assigned to this table fragments
     pub assigned_parallelism: TableParallelism,
+
+    /// The max parallelism specified when the streaming job was created, i.e., expected vnode count.
+    ///
+    /// The reason for persisting this value is mainly to check if a parallelism change (via `ALTER
+    /// .. SET PARALLELISM`) is valid, so that the behavior can be consistent with the creation of
+    /// the streaming job.
+    ///
+    /// Note that the actual vnode count, denoted by `vnode_count` in `fragments`, may be different
+    /// from this value (see `StreamFragmentGraph.max_parallelism` for more details.). As a result,
+    /// checking the parallelism change with this value can be inaccurate in some cases. However,
+    /// when generating resizing plans, we still take the `vnode_count` of each fragment into account.
+    pub max_parallelism: usize,
```

Comment:
Don't forget to expose this in our system table as well e.g. `src/frontend/src/catalog/system_catalog/rw_catalog/`. Ditto for any other new field in persisted metadata.

Reply:
Right. That's done in the following PR https://github.com/risingwavelabs/risingwave/pull/18672. 😁